packages/mds-policy-service/service/provider.ts

```diff
@@ -103,8 +111,31 @@ export const PolicyServiceProvider: ServiceProvider<PolicyService, PolicyService
     }),
   writePolicyIntentToPolicy: (context, intent_draft) =>
     serviceErrorWrapper('writePolicyIntentToPolicy', async () => {
-      // TODO insert ajv validation here
-      const { policy_id } = await PolicyRepository.writePolicy(translateIntentToPolicy(intent_draft))
+      const policy = translateIntentToPolicy(intent_draft)
+      switch (intent_draft.intent_type) {
+        case 'no_parking': {
+          validateNoParkingPolicyDomainModel(policy)
+        }
+        case 'permitted_vehicle_count': {
+          validatePermittedVehicleCountPolicyDomainModel(policy)
+          break
+        }
+        case 'parking_time_limit': {
+          validateParkingTimeLimitPolicyDomainModel(policy)
+          break
+        }
+        case 'permitted_parking': {
+          validatePermittedParkingPolicyDomainModel(policy)
+          break
+        }
+        default: {
+          validatePolicyDomainModel(policy)
+          break
+        }
+      }
```

Comment:
I feel like we should probably be validating the intents _before_ attempting to convert them to policies because they are:
a) Simpler objects
b) We want to verify their structure before doing any work with them (prior to the translation step), ideally, because there's a possibility that the caller of this method is using an out of date schema and may provide a malformed object (we've seen this before in dev with other services unfortunately and it has caused weirdness)
c) We really shouldn't have to validate at-runtime that the `translate` method is doing the correct thing, which is _essentially_ what this is doing. Verifying a translation of an input adds additional complexity to just verifying the input in-and-of itself.
d) If there _is_ an error, it might be something that the caller is unable to resolve, because perhaps it's a bug within `translateIntentToPolicy`, and not a bug within the Intent being sent in the first place. Any validation errors sent to an RPC client ought to be actionable.

Reply:
Okay, that's kind of a great point, yeah. 
So, you're saying we should be writing validators for this guy instead?
`IntentDraft<INTENT_TYPE>`

Reply:
@nessur precisely!

Reply:
I'd rather validate the final policy because some of those constants are vital to get correct, and they only become part of the policy object at the very end.

Reply:
Would there be any value to validating at both stages?

Reply:
I’m with @avatarneil on this one. Validate the intent at run-time and assert the full policy translation in unit tests.